### ISSUE TITLE
A mask conversion for cuDNN execution of scaled dot product attention should be fused

### ISSUE THREAD

**Issue by IvanYashchuk** [Lightning-AI/lightning-thunder#1789]
## 🚀 Feature

Currently, cuDNN doesn't support boolean masks for `torch.nn.functional.scaled_dot_product_attention` and this conversion is done eagerly at runtime:
https://github.com/Lightning-AI/lightning-thunder/blob/70c06666ee8847746f836d178aee541fe749c62f/thunder/executors/cudnnex.py#L286-L290

Instead `execution_transform` could be modified to do this conversion in Thunder so that it could be fused with previous operations in the trace
https://github.com/Lightning-AI/lightning-thunder/blob/70c06666ee8847746f836d178aee541fe749c62f/thunder/executors/cudnnex.py#L807
